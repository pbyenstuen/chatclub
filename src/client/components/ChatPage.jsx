import React from 'react'
import useLoader from "./useLoader";
import useWsChat from "./useWsChat";
import ErrorView from "./ErrorView";
import ChatView from './ChatView';

const ChatPage = ({ api }) => {
    const { data: user, loading, error, reload } = useLoader(async () => await api.auth.getUser());
    const { chatLog, sendMessage } = useWsChat();

    if (loading) {
        return <h2>Loading...</h2>
    }

    if (error) {
        return <ErrorView error={error} reload={reload} />;
    }

    return <ChatView
        user={user}
        chatLog={chatLog}
        sendMessage={sendMessage}
        api={api}
    />
}

export default ChatPage;
