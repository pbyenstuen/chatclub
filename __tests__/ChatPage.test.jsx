import React from "react";
import ReactDOM from "react-dom";
import { act, Simulate } from "react-dom/test-utils";
import { MemoryRouter } from "react-router";
import ChatPage from "../src/client/components/ChatPage";

let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

const mountComponent = async (component) => {
    await act(async () => {
        await ReactDOM.render(<MemoryRouter>{component}</MemoryRouter>, container);
    });
}

describe("ChatPage", () => {
    it("can show error message", async () => {
        const api = {
            auth: {
                getUser: () => {
                    throw new Error("Error")
                }
            }
        }

        await mountComponent(<ChatPage api={api} />);

        expect(container.innerHTML).toMatchSnapshot();
        expect(container.querySelector("div").textContent).toEqual("Something went wrong: Error: Error");
    });
});
