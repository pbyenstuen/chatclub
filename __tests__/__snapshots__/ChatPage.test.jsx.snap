// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatPage can show error message 1`] = `"<div>Something went wrong: Error: Error</div><button>TRY AGAIN</button>"`;
